sw/host/opentitanlib/src/util/num_de.rs

```diff
@@ -189,12 +187,27 @@ macro_rules! impl_parse_int_enc {
                 Self::from_str_radix(src, $radix).map_err(|e| e.into())
             }
         }
+
+        impl<T: ParseInt + $fmt> fmt::Display for $ty<T> {
+            fn fmt(&self, f: &mut fmt::Formatter) -> Result<(), fmt::Error> {
+                <dyn $fmt>::fmt(&self.0, f)
```

Comment:
Why the `dyn`...?

Reply:
`$fmt` is a trait object, without the `dyn` rustc complains.

Reply:
```
warning: trait objects without an explicit `dyn` are deprecated
   --> sw/host/opentitanlib/src/util/num_de.rs:193:18
    |
193 |                 <$fmt>::fmt(&self.0, f)
    |                  ^^^^ help: use `dyn`: `dyn $fmt`
...
208 | impl_parse_int_enc!(OctEncoded, 8, fmt::Octal);
    | ---------------------------------------------- in this macro invocation
    |
    = note: `#[warn(bare_trait_objects)]` on by default
    = warning: this is accepted in the current edition (Rust 2018) but is a hard error in Rust 2021!
    = note: for more information, see <https://doc.rust-lang.org/nightly/edition-guide/rust-2021/warnings-promoted-to-error.html>
    = note: this warning originates in the macro `impl_parse_int_enc` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Reply:
AFAICT it's equivalent to this https://godbolt.org/z/z4acx1zxT

Reply:
This is a diagnostics bug: https://github.com/rust-lang/rust/issues/96987